sample-tracking-status-overview-app/src/main/groovy/life/qbic/portal/sampletracking/datasources/samples/SamplesDbConnector.groovy

```diff
@@ -35,6 +35,41 @@ class SamplesDbConnector implements CountSamplesDataSource {
         this.connectionProvider = connectionProvider
     }
 
+    /**
+     * {@inheritDoc}
+     * @since 1.0.0
+     * We need to fetch all samples initially, since only the latest status counts. This only remains true, while
+     * Status.DATA_AVAILABLE is the last status a sample can reach.
+     */
+    @Override
+    List<String> fetchSampleCodesWithData(String projectCode) throws DataSourceException {
+        String queryTemplate = Query.fetchLatestSampleEntries()
```

Comment:
You would have to remove the `;` semicolon from every query to allow for modification first.
```suggestion
        String statusFilter = "WHERE <filter for sample status here>"
        String queryTemplate = Query.fetchLatestSampleEntries() + statusFilter + ";"
```